library/general/src/lib/ui/srv_status_component.rb

```diff
@@ -0,0 +1,236 @@
+
+    # @return [YaST::Term]
+    def widget
+      Frame(
+        _("Service Status"),
+        VBox(
+          ReplacePoint(Id("#{id_prefix}_status"), status_widget),
+          reload_widget,
+          VSpacing(),
+          on_boot_widget
+        )
+      )
+    end
+
+    # Handles the input triggered by the widgets, this method must be called in
+    # the event loop of the dialog using the component.
+    def handle_input(input)
+      case input
+      when "#{id_prefix}_stop"
+        stop_service
+        refresh_widget
+      when "#{id_prefix}_start"
+        start_service
+        refresh_widget
+      when "#{id_prefix}_reload"
+        @reload = Yast::UI.QueryWidget(Id(input), :Value)
+      when "#{id_prefix}_enabled"
+        @enabled = Yast::UI.QueryWidget(Id(input), :Value)
+        @enabled_callback.call(@enabled) if @enabled_callback
+      else
+        log.info "Input not handled by SrvStatusComponent: #{input}"
+      end
+    end
+
+    # Updates the widget to reflect the current status of the service and the
+    # settings
+    def refresh_widget
+      Yast::UI.ChangeWidget(Id("#{id_prefix}_reload"), :Enabled, service_running?)
+      Yast::UI.ChangeWidget(Id("#{id_prefix}_reload"), :Value, service_running? && @reload)
+      Yast::UI.ChangeWidget(Id("#{id_prefix}_enabled"), :Value, @enabled)
+      Yast::UI.ReplaceWidget(Id("#{id_prefix}_status"), status_widget)
+    end
+
+    # Reloads the service only if the user requested so. It should be called
+    # after saving the settings.
+    def reload
+      reload_service if service_running? && @reload
+    end
+
+    # rubocop:disable Style/TrivialAccessors
+
+    # Checks if the user requested the service to be enabled on boot
+    #
+    # @return [Boolean]
+    def enabled?
+      @enabled
+    end
+
+    # rubocop:enable Style/TrivialAccessors
+
+    # Content for the help
+    def help
+      _(
+        "<p><b><big>Current status</big></b><br>\n"\
+        "Displays the current status of the service. The status will remain "\
+        "the same after saving the settings, independently of the value of "\
+        "'start service during boot'.</p>\n"\
+        "<p><b><big>Reload After Saving Settings</big></b><br>\n"\
+        "Only applicable if the service is currently running. "\
+        "Ensures the running service reloads the new configuration after "\
+        "saving it (via 'ok' or 'save' buttons).</p>\n"\
+        "<p><b><big>Start During System Boot</big></b><br>\n"\
+        "Check this field to enable the service at system boot. "\
+        "Un-check it to disable the service. "\
+        "This does not affect the current status of the service in the already "\
+        "running system.</p>\n"
+      )
+    end
+
+  protected
+
+    attr_reader :id_prefix
+
+    # Checks if the service is currently running
+    #
+    # Must be redefined for services not following standard procedures
+    #
+    # @return [Boolean]
+    def service_running?
```

Comment:
that exactly show why Service API is not nice...this method should not be needed and instead it should call `service.running?`